File: 351002/Lashchenka/publisher/src/main/java/by/bsuir/distcomp/dto/response/EditorResponseTo.java
package by.bsuir.distcomp.dto.response;

import com.fasterxml.jackson.annotation.JsonIgnore;

public class EditorResponseTo {
    private Long id;
    private String login;
    private String password;
    private String firstname;
    private String lastname;
    private String role;

    public EditorResponseTo() {}

    public EditorResponseTo(Long id, String login, String password, String firstname, String lastname) {
        this.id = id;
        this.login = login;
        this.password = password;
        this.firstname = firstname;
        this.lastname = lastname;
    }

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public String getLogin() { return login; }
    public void setLogin(String login) { this.login = login; }
    @JsonIgnore
    public String getPassword() { return password; }
    public void setPassword(String password) { this.password = password; }
    public String getFirstname() { return firstname; }
    public void setFirstname(String firstname) { this.firstname = firstname; }
    public String getLastname() { return lastname; }
    public void setLastname(String lastname) { this.lastname = lastname; }
    public String getRole() { return role; }
    public void setRole(String role) { this.role = role; }
}

File: 351004/Zalivako/DistributedComputing/src/Application/DTOs/Responses/EditorResponseTo.cs
namespace Application.DTOs.Responses
{
    public class EditorResponseTo(
        string login,
        string password,
        string firstName,
        string lastName)
    {
        public long Id { get; set; }

        public string Login { get; set; } = login;

        public string Password { get; set; } = password;

        public string Firstname { get; set; } = firstName;

        public string Lastname { get; set; } = lastName;

        public string Role { get; set; } = string.Empty;
    }
}
